### PR TITLE
feat: achievement is seen

### PR DESCRIPTION
**Pull request by LucasFalkowsky**
partially closes: https://github.com/corona-school/project-user/issues/1029

What was done:

- mutation to update the isSeen field of user achievements was implemented

### CODE REVIEW

File: graphql/user/mutations.ts [hunk 1]
```diff
@@ -74,6 +75,16 @@ export class MutateUserResolver {
...
 
+    @Mutation(() => Boolean)
+    @Authorized(Role.USER)
+    async achievementIsSeen(@Arg('achievementId') achievementId: number) {
+        await prisma.user_achievement.update({
+            where: { id: achievementId },
+            data: { isSeen: true },
+        });
+        return true;
+    }
```
[the remark this answers is not in the document]
Comment (dhenkel92) [reply graphql/user/mutations.ts:86]: Or even `markAchievementAsSeen`

File: common/achievement/get.ts [hunk 2]
```diff
@@ -69,6 +69,7 @@ const assembleAchievementData = async (userAchievements: User_achievement[], use
     const newAchievement = state === achievement_state.COMPLETED && !userAchievements[resultIndex].isSeen;
 
     return {
+        id: userAchievements[resultIndex].id,
```
Comment (dhenkel92) [inline common/achievement/get.ts:72]: Wouldn't this result in an error if `resultIndex == null`?

Comment (LucasFalkowsky) [reply common/achievement/get.ts:72]: I removed result index since it made no sense.

**Review by LomyW — APPROVED** [review]
⚡️

**Pull request merged**